src/slacko.ml

```diff
@@ -765,33 +765,22 @@ let groups_list ?exclude_archived session =
       (match d |> groups_list_obj_of_yojson with
         | Result.Ok x -> `Success x.groups
         | Result.Error x -> `ParseFailure x)
-    | #bot_error
     | #parsed_auth_error as res -> res
     | _ -> `Unknown_error
 
-exception No_matches
-exception No_unique_matches
-exception Lookup_failed
-
 (* look up the id of query from results provided by the listfn *)
 let lookupk session listfn filterfn k =
   match%lwt listfn session with
-  | `Success channels -> (match List.filter filterfn channels with
-    | [] -> Lwt.fail No_matches
-    | [x] -> Lwt.return @@ k x
-    | _ -> Lwt.fail No_unique_matches)
-  | _ -> Lwt.fail Lookup_failed
+  | #parsed_auth_error as e -> Lwt.return e
+  | `Success items -> Lwt.return @@ k @@ List.filter filterfn items
```

Comment:
What if the lookup returns neither an `auth_error` nor a `Success`? It should probably have an `| _ -> ???` case with some defined behaviour.

I am not sure whether taking the "ensure there is one channel" logic out from here and the duplicaing it in every caller is a good idea. They all need to handle the case in probably the same way, so it might as well be done in one, common location.